### PR TITLE
remove all pre-existing upgrade listeners

### LINKED ISSUE #249: Node server crashes with ERR_INTERNAL_ASSERTION error
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.x

### Plugin version

7.1.3

### Node.js version

14.x

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

Azure AppServices for linux (Node running inside docker container)

### Description

We are getting some intermittent container crashes and looking at the logs looks like source of the exception is `@fastify/websockets` (logs below).  We are using @fastify in combination with [TRPC](https://github.com/trpc/trpc) subscriptions for real-time communication. 

```
Container exited unexpectedly: last 10 seconds logs [2023-04-17T06:26:19.633055587Z Error [ERR_INTERNAL_ASSERTION]: This is caused by either a bug in Node.js or incorrect usage of Node.js internals.
2023-04-17T06:26:19.633108688Z Please open an issue with this stack trace at https://github.com/nodejs/node/issues
2023-04-17T06:26:19.633116188Z 
2023-04-17T06:26:19.633120988Z     at new NodeError (internal/errors.js:322:7)
2023-04-17T06:26:19.633126088Z     at assert (internal/assert.js:14:11)
2023-04-17T06:26:19.633131288Z     at ServerResponse.assignSocket (_http_server.js:225:3)
2023-04-17T06:26:19.633136088Z     at Server. (/usr/src/app/server/dist/node_modules/@fastify/websocket/index.js:52:19)
2023-04-17T06:26:19.633141288Z     at Server.emit (events.js:400:28)
2023-04-17T06:26:19.633145788Z     at Server.emit (domain.js:475:12)
2023-04-17T06:26:19.633149988Z     at onParserExecuteCommon (_http_server.js:703:14)
2023-04-17T06:26:19.633154388Z     at onParserExecute (_http_server.js:616:3) {
2023-04-17T06:26:19.633159188Z   code: 'ERR_INTERNAL_ASSERTION'
2023-04-17T06:26:19.633164488Z }
]
```

I believe server should not completely crash and restart on such websocket errors. Any help or guidelines on we can resolve this problem. Not sure if this is the right place to raise the issue as the problem could be in the [fastify plugin for TRPC](https://github.com/fastify/fastify-plugin)? But from the error logs we can only pinpoint to `@fastify/websocket`.

Thanks

### Steps to Reproduce

We are unable to reproduce this issue in our development or staging environments, it is happening only in production.

### Expected Behavior

The server should not crash with an ERR_INTERNAL_ASSERTION error.

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #249 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -38,6 +38,7 @@ function fastifyWebsocket (fastify, opts, next) {
   const wss = new WebSocket.Server(wssOptions)
   fastify.decorate('websocketServer', wss)
 
+  websocketListenServer.removeAllListeners('upgrade')
```
Comment (climba03003) [inline index.js:41]: I am not sure if this is a correct approach.
Removing all the others listening is breaking changes and we remove all the ability for others to do anything on the `upgrade` event.
For example, they would like to log the event.

Instead, a proper fix should be inspect if `socket` already assigned.
Warn the developer they are mixing other `websocket` library and we are skipping the logic of that request.

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (airhorns) [reply index.js:41]: Yeah, I also feel like silently mutating the server is a bit unexpected. Is this is the only way to make it work? And if so, could we at least log a big deprecation warning or something that explains fastify-websocket doesn't play nice with any other libraries?

Comment (mcollina) [reply index.js:41]: I'm going to work on an alternative approach.

**Comment by mcollina** [comment]
This is a complete rewrite of the shutdown logic to avoid monkey-patching. It's a semver-major change.
This also includes a behavior change now, as websocket requests received while the server is closing will now receive a 503 like all other requests.

**Review by climba03003 — APPROVED** [review]

**Pull request merged**